WalletWasabi.Fluent/Helpers/LabelHelpers.cs

```diff
@@ -11,10 +11,14 @@ public static IEnumerable<(string Label, int Score)> GetLabelsWithRanking(this W
 	{
 		var labelPool = new Dictionary<string, int>(); // int: score.
 
-		// Make recent and receive labels count more for the current wallet
+		var labelsByWalletId = WalletHelpers.GetLabelsByWallets();
+
+		// Make recent and receive labels count more for the current wallet.
 		var multiplier = 100;
-		foreach (var label in wallet.KeyManager.GetReceiveLabels().Reverse().SelectMany(x => x))
+		var currentWalletReceiveLabels = labelsByWalletId.First(x => x.WalletId == wallet.WalletId).ReceiveLabels;
```

Comment:
What if there are no labels? `First` will throw.

Reply:
Yes, I misunderstand the code. You will do another enumeration right before, so the wallet must be in the array. Nevermind.